fvm/evm/emulator/state/delta.go

```diff
@@ -99,6 +99,21 @@ func (d *DeltaView) Exist(addr gethCommon.Address) (bool, error) {
 
 // CreateAccount creates a new account for the given address
 func (d *DeltaView) CreateAccount(addr gethCommon.Address) error {
+	// If a address already exists the balance is carried over to the new account.
+	// Carrying over the balance ensures that Ether doesn't disappear. (legacy behaviour of the Geth stateDB)
+	exist, err := d.Exist(addr)
```

Comment:
I wasn't sure about this before on the first PR so I wanted to wait to see how it's all implemented, but seeing this now it makes me again cautious about what happens when you suicide an account, I believe funds are being destroyed, I believe acording to the eip the funds should be recovered https://eips.ethereum.org/EIPS/eip-6780

Reply:
That EIP is not finalized yet, while it's claimed that funds are transferred, I found this part of the code to anyway reset the balance.  https://github.com/ethereum/go-ethereum/blob/master/core/state/statedb.go#L456

Reply:
That logic is only if you call create account and the hasn't been flagged with selfdestruct.

Reply:
I updated the logic to work as you described it.